### PR TITLE
CASMINST-4362, CASMINST-4161, CASMINST-4183 adjust interface mac bond…

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

## Summary and Scope

Backports a fix for the interface MAC comparison test from 1.2

## Issues and Related PRs

* Resolves CASMINST-4362

### Tested on:

  * `#surtur`

### Test description:

```
....
....

Total Tests: 51, Total Passed: 51, Total Failed: 0, Total Execution Time: 58.6371 seconds
surtur-ncn-m001-pit:/var/www/ephemeral/jsalmela #
```

## Risks and Mitigations

Low.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: goss-testing/scripts/check_interface_mac_matches_bond.sh [hunk 1]
```diff
@@ -61,67 +51,49 @@ Exits 0 if this is the case. Exits non-0 if not, or if there are any errors.
...
-# Validate that we actually got a MAC address for each
-[[ -n $INT_MAC ]] || error 70 "No MAC address found for $INTERFACE"
-[[ -n $BND_MAC ]] || error 75 "No MAC address found for bond0"
-
-# Finally, validate that they match
-[[ $INT_MAC == $BND_MAC ]] || error 90 "MAC address of $INTERFACE does not match MAC address of bond0"
+# if the interface does not show, it might be the option CAN
+if ! eval ip addr show dev "$INTERFACE" 1>/dev/null; then
+  if [[ "$INTERFACE" == "bond0.can0" ]]; then
+    echo "$INTERFACE is optional...skipping"
```
Comment (heemstra) [inline goss-testing/scripts/check_interface_mac_matches_bond.sh:75]: This is not accurate in csm-1.0.

**Review by heemstra — CHANGES_REQUESTED** [review]
If bond0.can0 doesn't exist it should be a failure in csm-1.0

File: goss-testing/scripts/check_interface_mac_matches_bond.sh (hunk 1 above)
Comment (jacobsalmela) [reply goss-testing/scripts/check_interface_mac_matches_bond.sh:75]: I'll adjust this

Comment (jacobsalmela) [reply goss-testing/scripts/check_interface_mac_matches_bond.sh]: Fixed to fail if any interface this is checking for is missing.

**Comment by jacobsalmela** [comment]
Not needed since the commit that broken this was reverted

**Pull request closed without merging**